ompi/mca/osc/rdma/osc_rdma.h

```diff
@@ -673,4 +673,84 @@ static inline mca_btl_base_am_rdma_module_t *ompi_osc_rdma_selected_am_rdma(ompi
     return module->alternate_am_rdmas[btl_index];
 }
 
+/**
+ * @brief Copies from src to dest buffers that may or may not be accelerator
+ *        buffers
+ *
+ * @param[in]  dest           destination buffer
+ * @param[in]  src            source buffer
+ * @param[in]  size           length to copy
+ *
+ * @returns dest
+ */
+static inline void *osc_rdma_accelerator_mem_copy(void *dest, const void *src, size_t size)
+{
+    int res, dev_id;
+    uint64_t flags;
+    int dest_is_accel = opal_accelerator.check_addr(dest, &dev_id, &flags);
+    int src_is_accel = opal_accelerator.check_addr(src, &dev_id, &flags);
+
+    if (dest_is_accel == 0 && src_is_accel == 0) {
+        return memcpy(dest, src, size);
+    }
+
+    res = opal_accelerator.mem_copy(MCA_ACCELERATOR_NO_DEVICE_ID, MCA_ACCELERATOR_NO_DEVICE_ID,
+                                  dest, src, size, MCA_ACCELERATOR_TRANSFER_UNSPEC);
+    if (OPAL_SUCCESS != res) {
+        opal_output(0, "Error in accelerator memcpy");
+        abort();
+    } else {
+        return dest;
+    }
+}
+
+/**
+ * @brief Moves memory from src to dest buffers that may or may not be accelerator
+ *        buffers
+ *
+ * @param[in]  dest           destination buffer
+ * @param[in]  src            source buffer
+ * @param[in]  size           length to copy
+ *
+ * @returns dest
+ */
+static inline void *osc_rdma_accelerator_mem_move(void *dest, const void *src, size_t size)
+{
+    int res, dev_id;
+    uint64_t flags;
+    int dest_is_accel = opal_accelerator.check_addr(dest, &dev_id, &flags);
+    int src_is_accel = opal_accelerator.check_addr(src, &dev_id, &flags);
+
+    if (dest_is_accel == 0 && src_is_accel == 0) {
+        return memmove(dest, src, size);
+    }
+
+    res = opal_accelerator.mem_move(MCA_ACCELERATOR_NO_DEVICE_ID, MCA_ACCELERATOR_NO_DEVICE_ID,
+                                  dest, src, size, MCA_ACCELERATOR_TRANSFER_UNSPEC);
+    if (OPAL_SUCCESS != res) {
+        opal_output(0, "Error in accelerator memmove");
+        abort();
+    } else {
+        return dest;
+    }
+}
+/**
+ * @brief Identifies whether the provided buffer is an accelerator buffer.
+ *
+ * @param[in]  buf            buffer
+ *
+ * @returns true              If buffer is accelerator buffer
+ * @returns false             If buffer is not accelerator buffer
+ */
+static inline bool osc_rdma_is_accel(void *buf)
+{
+    int dev_id;
+    uint64_t flags;
+    int buf_is_accel = opal_accelerator.check_addr(buf, &dev_id, &flags);
+    if (0 > buf_is_accel) {
+        /* This is not a recoverable error */
+        abort();
+    }
```

Comment:
same here. I understand that it is not a recoverable error, but I would argue that it still up the error handler setting on how to deal with that.